xbmc/addons/addoninfo/AddonType.cpp

```diff
@@ -56,3 +63,8 @@ bool CAddonType::IsDependencyType(TYPE type)
 {
   return dependencyTypes.find(type) != dependencyTypes.end();
 }
+
+bool CAddonType::IsVisibleInDialog(const std::shared_ptr<IAddon>& addon)
```

Comment:
Maybe call this `IsVisibleAddonType`?

Reply:
"Visible" still implies GUI. We need better wording. Could it be something like "Private". Don't know - not a native English speaker here.

Reply:
Maybe the definition of that "property" here is the wrong place. Why cant that logic be in the dialog?

Reply:
Fair point. I hadn't noticed invisibleTypes was defined in the PR. I think just inline this yet where it is used.

Reply:
to be honest, it is only in there because i saw the other `std::set` in there...
you're right, doesnt really belong there

Reply:
Hopefully the day will come when the base and GUI in Kodi are independent.
So it's always good to keep the GUI stuff out there.

My dream would be a sandbox system like in Chromium where the respective things are completely independent of each other and it doesn't matter if a Python or binary addon crashes.

Reply:
don‘t worry alwin, i‘m actually turning it upside down 🙃